### PR TITLE
ZRZ-531 - Replace fetch with manager.fetch

### PR DESCRIPTION
**Pull request by omarmosid**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -79,47 +80,47 @@ const sendEvent = async (
...
+        const msSinceLastEvent = Date.now() - parseInt(client.get('let') || '0') // _let = "_lastEventTime"
+        if (msSinceLastEvent > 1000) {
+          sendEvent('user_engagement', manager, event, settings)
+          client.set('engagementPaused', Date.now().toString())
+        }
       }
     }
-  }
 
-export default async function (manager: Manager, settings: ComponentSettings) {
```
Comment (jonnyparris) [inline src/index.ts:109]: Why did this need moving to encapsulate the onVibilityChange handler?

**Review by jonnyparris — APPROVED** [review]
Primary change for the manager fetch looks good but there's some extra noise in the diff that seems unnecessary. Let's keep it clean then merge, unless I've missed something about that engagement time handler?

File: src/index.ts (hunk 1 above)
Comment (bjesus) [reply src/index.ts:109]: So we have access to the `manager` object without needing to pass it around all the time.

Comment (jonnyparris) [reply src/index.ts:109]: but only the `sendEvent` function is using it and it hasn't been included here...at least, not yet! It's a good idea if that was the intention

**Review by jonnyparris — CHANGES_REQUESTED** [review]
see comment

File: src/index.ts (hunk 1 above)
Comment (jonnyparris) [reply src/index.ts:109]: aha spotted the `manager` in the onVisibilityChange handler, but indeed if sendEvent is there too then there's no need to inject it as an arg

**Review by jonnyparris — APPROVED** [review]

**Pull request merged**